tools/idf_monitor_base/serial_handler.py

```diff
@@ -115,8 +115,18 @@ def handle_serial_input(self, data, console_parser, coredump, gdb_helper, line_m
         ))
         if self._last_line_part != b'' and force_print_or_matched:
             self._force_line_print = True
+            if self._decode_panic != PANIC_DECODE_DISABLE:
+                if self._reading_panic == PANIC_READING:
+                    self._panic_buffer += self._last_line_part.replace(b'\r', b'')
+
             self.logger.print(self._last_line_part)
             self.logger.handle_possible_pc_address_in_line(self._last_line_part)
+            if gdb_helper:
+                self.check_panic_decode_trigger(self._last_line_part, gdb_helper)
+            with coredump.check(line):
```

Comment:
Hi @bugadani, is it meant to be `self._last_line_part` instead of `line` here? The variable `line` will be in many cases uninitialized.

Reply:
Hi @sio13

Thanks for catching it, it's definitely an error! On the other hand, I'm not sure if much of the patch is necessary. Let me check, and I'll update it accordingly.